### ISSUE TITLE
Build error on SSR Cloudflare

### ISSUE THREAD

**Issue by thiti-y** [withastro/astro#7664]
### What version of `astro` are you using?

2.8.2

### Are you using an SSR adapter? If so, which one?

Cloudflare

### What package manager are you using?

npm

### What operating system are you using?

Mac

### What browser are you using?

Chrome

### Describe the Bug

I use mongodb on SSR route, I have builded error with message in photo below.

<img width="683" alt="image" src="https://github.com/withastro/astro/assets/7018802/7c543207-0247-4e1e-8b14-8f18def1bcd9">

### What's the expected result?

Build success.

### Link to Minimal Reproducible Example

No

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by ematipico** [comment]
Cloudflare environment is not node. Remove the `require` and use ESM.
If those warnings come from a library, that library can't be used on cloudflare.

**Comment by thiti-y** [comment]
> Cloudflare environment is not node. Remove the `require` and use ESM. If those warnings come from a library, that library can't be used on cloudflare.

You mean that I don't use mongodb library in cloudflare workers, Right?

**Comment by ematipico** [comment]
> > Cloudflare environment is not node. Remove the `require` and use ESM. If those warnings come from a library, that library can't be used on cloudflare.
> 
> You mean that I don't use mongodb library in cloudflare workers, Right?

Exactly. Or check if they have an ESM version. Maybe that could help.

**Comment by bluwy** [comment]
I don't think mongodb would ever work in Cloudflare workers, unless you polyfill the nodejs modules with a vite plugin perhaps. But:

1. You'd get a larger server bundle size (Cloudflare has restrictions on that)
2. Polyfills run slower
3. Might not work correctly

So it's probably best to deploy to a Node.js server instead if you want to stick with mongodb. Closing as there's not much we can do here.
